pilot/pkg/security/authz/builder/testdata/tcp/deny-both-http-tcp-out.yaml

```diff
@@ -310,83 +329,99 @@ typedConfig:
                         safeRegex:
                           googleRe2: {}
                           regex: .*/ns/not-ns/.*
-                  - authenticated:
-                      principalName:
+                  - filterState:
+                      key: io.istio.peer_principal
+                      stringMatch:
                         safeRegex:
                           googleRe2: {}
                           regex: .*/ns/.*not-ns-suffix/.*
-                  - authenticated:
-                      principalName:
+                  - filterState:
+                      key: io.istio.peer_principal
+                      stringMatch:
                         safeRegex:
                           googleRe2: {}
                           regex: .*/ns/not-ns-prefix.*/.*
-                  - authenticated:
-                      principalName:
+                  - filterState:
+                      key: io.istio.peer_principal
+                      stringMatch:
                         safeRegex:
                           googleRe2: {}
                           regex: .*/ns/.*/.*
             - orIds:
                 ids:
-                - authenticated:
-                    principalName:
+                - filterState:
+                    key: io.istio.peer_principal
+                    stringMatch:
                       exact: spiffe://principal
-                - authenticated:
-                    principalName:
+                - filterState:
+                    key: io.istio.peer_principal
+                    stringMatch:
                       safeRegex:
                         googleRe2: {}
                         regex: spiffe://.*principal-suffix
-                - authenticated:
-                    principalName:
+                - filterState:
+                    key: io.istio.peer_principal
+                    stringMatch:
                       prefix: spiffe://principal-prefix
-                - authenticated:
-                    principalName:
+                - filterState:
+                    key: io.istio.peer_principal
+                    stringMatch:
                       safeRegex:
                         googleRe2: {}
                         regex: .+
             - notId:
                 orIds:
                   ids:
-                  - authenticated:
-                      principalName:
+                  - filterState:
+                      key: io.istio.peer_principal
+                      stringMatch:
                         exact: spiffe://not-principal
-                  - authenticated:
-                      principalName:
+                  - filterState:
+                      key: io.istio.peer_principal
+                      stringMatch:
                         safeRegex:
                           googleRe2: {}
                           regex: spiffe://.*not-principal-suffix
-                  - authenticated:
-                      principalName:
```

Comment:
I believe proxyless gRPC is also using this - grpc/authz/rbac_translator.go and equivalent in the other languages, this will break them. 

Wither generate new name only if generator is not gRPC, or maybe change the envoy filter that handles 'principalName' to transparently check the io.istio.peer_principal and fall back to socket-level ?

Reply:
gRPC is using Envoy RBAC? I find it that surprising since it's been changed multiple times in Envoy without any gRPC breakage. Can you point to where gRPC programs RBAC and why are no tests failing?

Reply:
So it seems that gRPC forked the definition of RBAC so changing Envoy will not actually help. I'll guard it instead for Envoy only.

Reply:
Thanks for raising this, I guarded it for gRPC. Envoy definition has diverged from gRPC so they will need to be aligned at some point in the future.

Reply:
Great, thanks